src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/readers/RecursiveTemplateResourceParser.java

```diff
@@ -108,10 +108,30 @@ protected void validateDepth() {
 
     private void addFileWatcher(Resource resource) {
         if (this.watchers == null) this.watchers = new ArrayList<>();
-        this.watchers.add(new FileWatcher<>(resource));
+        this.watchers.add(new IncludedTemplateWatcher(resource));
     }
 
     public List<FileWatcher<Object>> getWatchers() {
         return watchers;
     }
+
+    private class IncludedTemplateWatcher extends FileWatcher<Object> {
```

Comment:
So it updates the current watcher last check instead of its own. Makes sense, could you add a bit of javadoc explaining?

Reply:
done